astropy/time/formats.py

```diff
@@ -368,6 +369,7 @@ def value(self):
         raise NotImplementedError
 
     @classmethod
+    @functools.lru_cache(maxsize=1)
```

Comment:
Since almost always `pattern='*'`, I wonder if a route to an even larger speed-up would be to have as the very first part
```
if pattern == '*':
    return cls.subfmts
```

Reply:
That sounds fine, but I would put this after the validation of `pattern` as `str`.

Reply:
My suggestion was to *replace* the cache with that check, since I thought that might make the most common case even faster. Though if that is not the case -- the cache might just be a quick dict lookup -- I would stick with the cache and not complicate the code.

Reply:
Yeah, maybe just go with the cache. The only problem could come if `cls.subfmts` changes between calls, which seems like it should not happen ever.

Reply:
> Since almost always `pattern='*'`, I wonder if a route to an even larger speed-up would be to have as the very first part
> 
> ```
> if pattern == '*':
>     return cls.subfmts
> ```

Good idea - this is indeed at least as fast as doing the caching, and probably marginally safer just in case `cls.subfmts` does change (which it perhaps could do in some edge cases within a python session?).